api/app/signals/apps/api/views/attachment.py

```diff
@@ -61,3 +62,27 @@ def get_signal(self):
         pk = self.kwargs.get('parent_lookup__signal__pk')
         signal = get_object_or_404(Signal.objects.filter_for_user(self.request.user), pk=pk)
         return signal
+
+    def destroy(self, *args, **kwargs):
+        user = self.request.user
+        signal = self.get_signal()
+        attachment = self.get_object()
+
+        if signal.is_parent and not user.has_perm('signals.delete_attachment_of_parent_signal'):
```

Comment:
Please move the permissions to the correct place. For example see the `SIAUserPermissions` and the `SignalViewObjectPermission`.

The permission checks should live in a permission service for attachments. And then a special Permission class can be used to check the permissions for this action.

Reply:
Done

Reply:
As discussed we will defer this change to its own PR.